axelor-sale/src/main/java/com/axelor/apps/sale/service/ProductRestServiceImpl.java

```diff
@@ -64,25 +69,42 @@ public ProductRestServiceImpl(
       UserService userService,
       ProductRestService productRestService,
       AppBaseService appBaseService,
-      ProductPriceService productPriceService) {
+      ProductPriceService productPriceService,
+      UnitConversionService unitConversionService) {
     this.appSaleService = appSaleService;
     this.companyService = companyService;
     this.userService = userService;
     this.partnerRepository = partnerRepository;
     this.productRestService = productRestService;
     this.appBaseService = appBaseService;
     this.productPriceService = productPriceService;
+    this.unitConversionService = unitConversionService;
   }
 
   protected List<PriceResponse> fetchProductPrice(
-      Product product, Partner partner, Company company, Currency currency) throws AxelorException {
-
+      Product product, Partner partner, Company company, Currency currency, Unit unit)
+      throws AxelorException {
     List<PriceResponse> priceList = new ArrayList<>();
 
     BigDecimal priceWT =
         productPriceService.getSaleUnitPrice(company, product, false, partner, currency);
     BigDecimal priceATI =
         productPriceService.getSaleUnitPrice(company, product, true, partner, currency);
+    if (product.getSalesUnit() == null && product.getUnit() == null) {
+      throw new AxelorException(
+          TraceBackRepository.CATEGORY_NO_VALUE,
+          I18n.get(SaleExceptionMessage.PRODUCT_SALE_UNIT_IS_NULL),
+          product.getId());
+    }
+    Unit convertFrom = product.getSalesUnit() != null ? product.getSalesUnit() : product.getUnit();
```

Comment:
You can put this line inside the unit block since we only use it if we put an unit in the request body.